### PR TITLE
Do something about the volume of NCI records

### LINKED ISSUE #113: Do something about the volume of NCI records
The NCI importer pulls somewhere on the order of hundreds of thousands of entries (I let it run for a few hours and it still hadn't finished so I stopped it manually). This is pretty far off from previous DGIdb entries (which were like ~6k interactions). If this is correct behavior, then it's correct, but it's a pretty unwieldy size either way.

### PR DESCRIPTION
**Pull request by mcannon068nw**
Fixes #113

### CODE REVIEW

**Comment by mcannon068nw** [comment]
recovered TSV added to s3

File: server/lib/genome/importers/file_importers/nci.rb [hunk 1]
```diff
@@ -1,125 +1,55 @@
...
     attr_reader :file_path
-    attr_reader :source_db_name
-    attr_reader :source
-    attr_reader :drug_filter
+    attr_reader :worksheet
 
     def initialize(file_path)
-      @file_path = handle_file_location file_path
-      @source_db_name = 'NCI'
+      @file_path = file_path
```
Comment (jsstevenson) [inline server/lib/genome/importers/file_importers/nci.rb:7]: should be `@file_path = handle_file_location file_path`

File: server/lib/genome/importers/file_importers/nci.rb [hunk 2]
```diff
@@ -1,125 +1,55 @@
...
-          source_db_name: source_db_name,
-          full_name: 'NCI Cancer Gene Index',
-          license: 'Public domain',
-          license_link: 'https://www.cancer.gov/policies/copyright-reuse'
-        }
+          {
+              base_url: 'https://wiki.nci.nih.gov/display/cageneindex/The+Cancer+Gene+Index+Gene-Disease+and+Gene-Compound+XML+Documents',
+              site_url: 'https://wiki.nci.nih.gov/display/cageneindex',
+              citation: 'https://wiki.nci.nih.gov/display/cageneindex',
+              source_db_version:  get_version,
```
Comment (jsstevenson) [inline server/lib/genome/importers/file_importers/nci.rb:33]: `get version` should be deleted and we should be hardcoding the data date. Last update on dgidb is listed as "14-September-2017" so we can probably use that

**Review by jsstevenson — CHANGES_REQUESTED** [review]

**Comment by jsstevenson** [comment]
Oh, one other thing

in lib/tasks/importers.rake, you should delete lines 136 and 137 

```
      # TODO: remove WIP importers
      adj_file_importer_names = file_importer_names.reject { |name| name == 'nci' }
```

so that NCI gets added back into the working importers list

File: server/lib/genome/importers/file_importers/nci.rb [hunk 3]
```diff
@@ -1,125 +1,55 @@
...
     end
 
-    def run_parser
-      @drug_filter = DrugFilter.new
-      parser = Nokogiri::XML::SAX::Parser.new(drug_filter)
-      parser.parse(File.open(@file_path))
+    def import
+      remove_existing_source
+      create_new_source
+      create_interaction_claims
```
Comment (jsstevenson) [inline server/lib/genome/importers/file_importers/nci.rb:18]: this method name should be changed to `create_claims` and the `remove_existing_source` and `create_new_source` lines should be deleted

**Review by jsstevenson — CHANGES_REQUESTED** [review]

File: server/lib/genome/importers/file_importers/nci.rb [hunk 4]
```diff
@@ -1,125 +1,52 @@
...
-        gene_claim = create_gene_claim(record[0],'CGI Gene Name')
-        drug_claim = create_drug_claim(record[1],record[1],'NCI Drug Name')
-        create_drug_claim_alias(drug_claim, record[2], 'NCI Drug Code')
-        interaction_claim = create_interaction_claim(gene_claim, drug_claim)
-        create_interaction_claim_publication(interaction_claim, record[3])
-        create_interaction_claim_link(interaction_claim, 'The Cancer Gene Index Gene-Disease and Gene-Compound XML Documents', 'https://wiki.nci.nih.gov/display/cageneindex/The+Cancer+Gene+Index+Gene-Disease+and+Gene-Compound+XML+Documents' )
-        end
-      backfill_publication_information
+    def remove_existing_source
+      Utils::Database.delete_source('NCI')
```
Comment (jsstevenson) [inline server/lib/genome/importers/file_importers/nci.rb:21]: this method isn't used, can be deleted

File: server/lib/genome/importers/file_importers/nci.rb [hunk 5]
```diff
@@ -1,125 +1,52 @@
...
     def initialize(file_path)
       @file_path = handle_file_location file_path
-      @source_db_name = 'NCI'
     end
 
-    def default_filetype
-      'xml'
+    def get_version
+      source_db_version = Date.today.strftime('%d-%B-%Y')
+      @new_version = source_db_version
```
Comment (jsstevenson) [inline server/lib/genome/importers/file_importers/nci.rb:11]: no longer used, can be deleted

**Review by jsstevenson — CHANGES_REQUESTED** [review]

**Review by jsstevenson — APPROVED** [review]

**Pull request merged**